src/olympia/addons/tasks.py

```diff
@@ -937,19 +938,23 @@ def extract_colors_from_static_themes(ids, **kw):
 @task
 @use_primary_db
 def recreate_theme_previews(addon_ids, **kw):
-    from olympia.versions.tasks import generate_static_theme_preview
-
     log.info('[%s@%s] Recreating previews for themes starting at id: %s...'
              % (len(addon_ids), recreate_theme_previews.rate_limit,
                 addon_ids[0]))
     addons = Addon.objects.filter(pk__in=addon_ids).no_transforms()
+    only_missing = kw.get('only_missing', False)
 
     for addon in addons:
-        log.info('Recreating previews for theme: %s' % addon.id)
         version = addon.current_version
         if not version:
             continue
         try:
+            if only_missing:
+                with_size = (VersionPreview.objects.filter(version=version)
+                             .exclude(sizes={}).count())
+                if with_size == 3:
```

Comment:
```suggestion
                if with_size == `len(amo.THEME_PREVIEW_SIZES)`
```

Reply:
yeah... I was concerned about `len(amo.THEME_PREVIEW_SIZES)` reducing from 3 (the original plan last year was to standardize on 1 size and drop the others).  But I guess it's better the task code changes and so tests will fail and hopefully trigger us to look into it more.